### ISSUE TITLE
camelcase rule with {properties: never} shouldn't check assignment

### ISSUE THREAD

**Issue by kentor** [eslint/eslint#2189]
In the docs, with the `{ "properties": "never" }` option to `camelcase` it says:

   Properties can have the following values:

1. always is the default and checks all proprty (typo btw) names
1. never does not check property names at all

However it does check when defining a property outside of the literal:

```yaml
---
rules:
  camelcase: [2, {properties: never}]
```

```javascript
var obj = {
  a_a: 1
};

obj.a_b = 2;
```

```
index.js
  5:4  error  Identifier 'a_b' is not in camel case  camelcase

✖ 1 problem (1 error, 0 warnings)
```
